homeassistant/components/smart_meter_texas/get_ssl_context.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import re
+import socket
+import ssl
+import urllib
+
+import OpenSSL.crypto as crypto
+import certifi
+from smart_meter_texas.const import BASE_HOSTNAME
+
+_LOGGER = logging.getLogger(__name__)
+
+def get_ssl_context():
```

Comment:
This needs to live in the pypi package per https://developers.home-assistant.io/docs/development_checklist/

Reply:
Would including the function within the smart_meter_texas library be acceptable?  
If so, @grahamwetzler  do you have an opinion on where this logic should live?

Reply:
> Would including the function within the smart_meter_texas library be acceptable?

Yes 👍

Reply:
I think it’s fine to move into `smart_meter_texas/__init__.py` for now.

Reply:
Sounds good. 

I'll create a static class named ClientSSLContext and put the get_ssl_context method in there, along with any helper methods needed for getting the CA Issuer URI via the asn.1 data.  I'll work on getting the asyncio/executor logic working with this change as well.

I will keep it so the Client uses a standard SSL Context by default unless one is passed in explicitly during initialization. 

This sure is turning out to be a fun and educational exercise with python. 😅

I greatly appreciate all the help, feedback, and patience!